provider/server/machine_list_test.go

```diff
@@ -57,13 +67,44 @@ var _ = Describe("ListMachine", func() {
 			return libvirt.DomainState(domainState)
 		}).Should(Equal(libvirt.DomainRunning))
 
-		By("List machines")
-		resp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+		By("listing machines using machine Id")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).NotTo(BeEmpty())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return listResp.Machines[0].Status.State
+		}).Should(Equal(iri.MachineState_MACHINE_RUNNING))
+
+		By("listing machines using correct Label selector")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
```

Comment:
Should this still be within the Eventually block? I believe that since the `listing machines using machine Id` step has already succeeded, the machine states are now stable.

Reply:
Eventually block removed